Upstream merge

walksanatora opened the pull request:
## Short description
pulls upstream content as of CL https://discord.com/channels/310555209753690112/1079439851750953082/1411742690806333489

## Why we need to add this
upstream content good!

## Media (Video/Screenshots)

## Checks

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work. "work" as in I can join the server and go to lobby.
- [ ] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
:cl: walksanatora
- add: Merged upstream content

Winkarst-cpu committed:
SharedKitchenSpikeSystem bugfixes (#39959)

* Fixes

* Update

* Update

Kowlin committed:
Stop derelict borgs from duplicating their ghost roles. (#39992)

Add reregister to derelict borgs

PJBot committed:
Automatic changelog update

iaada committed:
Clown bags squeak when inserting items (#39931)

squeak!

Co-authored-by: iaada <iaada@users.noreply.github.com>

PJBot committed:
Automatic changelog update

ScarKy0 committed:
Bagel AI Turrets + Camera Coverage (#39968)

PJBot committed:
Automatic changelog update

ScarKy0 committed:
Marathon AI Turrets + Camera Coverage (#39969)

PJBot committed:
Automatic changelog update

ScarKy0 committed:
Nullable messydrinker tag (#40002)

init

github-actions[bot] committed:
Update Credits (#40005)

Co-authored-by: PJBot <pieterjan.briers+bot@gmail.com>

ToastEnjoyer committed:
Removed unused asset from devmap (#39974)

SlamBamActionman committed:
Fix latejoinspawner on Elkridge (#40010)

SlamBamActionman committed:
Fix latejoinspawner on Exo (#40011)

SlamBamActionman committed:
Fix latejoin spawner on Plasma (#40012)

ArtisticRoomba committed:
Staging to master merge (#40013)

AndrewFenriz committed:
Migrate all mechs to PartAssembly and remove legacy MechAssemblySystem (#39027)

* Removed the old MechAssembly system and component.
Converted all mechs to use the unified PartAssembly system.
Removed dismantling mechs during assembly logic to simplify the code.

* Delete Chassis via migration

PJBot committed:
Automatic changelog update

VasilisThePikachu committed:
Revert "Admin Log Browser Improvements (#39130)"

This reverts commit f67cebf7a4ed451f1911aa5c8bf6f04c198b917f.

Per request of @Kowlin and @southbridge-fur

Check out https://github.com/space-wizards/space-station-14/issues/39960 for further information

slarticodefast committed:
Stable merge (#40025)

slarticodefast committed:
improve spawnpoint error logging (#40021)

iaada committed:
TriggerOnMeleeHit and more (#39826)

Co-authored-by: iaada <iaada@users.noreply.github.com>
Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

spanky-spanky committed:
Terminal Hotfix (#40027)

* Resave terminal.yml

* Fix airlock

slarticodefast committed:
Merge stable into master (#40034)

Travis-G-Reid committed:
Fix docstring typo starts -> stops (#40031)

Commit:
set slot priority to 4

SlamBamActionman committed:
Moths cannot eject items from military boots (#40049)

PJBot committed:
Automatic changelog update

slarticodefast committed:
Predict InjectorSystem (#39976)

* predict injectors

* hide verbs if no options

PJBot committed:
Automatic changelog update

SlamBamActionman committed:
Fix incorrect bullet & cartridge names (#39993)

* Fix conflicting names

* zero zeroes

insoPL committed:
Texture Scaling for clothing (#39714)

scale

FairlySadPanda committed:
Lizard Tails Can Be Hidden By Clothing (#40026)

Taken from @TiniestShark's PR.

Nox38 committed:
Fixed Corpsman Name (finally) (#40055)

Fixed Corpsman name (finally)

Signed-off-by: Nox38 <nebulousnox38@gmail.com>

iaada committed:
no utensil

southbridge-fur committed:
Event Shuttle Fixes (#40059)

PJBot committed:
Automatic changelog update

IProduceWidgets committed:
Update 4 visitor shuttles & nanomed inventories (#39718)

PJBot committed:
Automatic changelog update

M4rchy-S committed:
Fix radiation vomit for dead mobs (#40020)

* Fix Radiation Vomit for dead mobs

* Update Content.Server/Destructible/Thresholds/Behaviors/VomitBehavior.cs

Co-authored-by: lzk <124214523+lzk228@users.noreply.github.com>

* Fix Radiation Vomit for dead mobs

* Fix Radiation Vomit system for dead mobs

* refactors

* Adding mobStateSystem for validation

* refactor

* Unrelated cleanup

---------

Co-authored-by: lzk <124214523+lzk228@users.noreply.github.com>
Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

PJBot committed:
Automatic changelog update

walksanatora committed:
merge 'wizden/stable' into 'walksanatora/upstream-merge'

walksanatora committed:
and this kids is why we dont merge upstream PRs until they are closed.

walksanatora committed:
oh god there is alot of USSP uplink errors... I dont think I have the battery life for this.

walksanatora committed:
find and replace saves the day

MilenVolf committed:
Remove empty `drink-component.ftl` file (#40064)

Probably had to be removed in #39031

Princess-Cheeseballs committed:
Add myself to Codeowners for Stunnable and Nutrition (#40061)

* I probably should do this

* Can't cut it up that easy cause of the events. Guess I'm just subscribed to all nutrition stuff ;_;

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

IProduceWidgets committed:
prevent double-mapping lights (#39939)

works?

SolidSyn committed:
Adds a secHUD to the noir-tech glasses (#39859)

* Adds a SecHUD to the noir-tech glasses

* Adds a Security HUD to the noir-tech glasses

* Revert "Adds a Security HUD to the noir-tech glasses"

This reverts commit 68d7b9b6e190618e44a94df71cf311e056892392.

* Adds a security HUD to the noir-tech glasses

PJBot committed:
Automatic changelog update

walksanatora committed:
appease yaml linter.

walksanatora committed:
Revert "Prevented Engiborgs from picking up AI lawboards (#39730)"

This reverts commit 1800be1f5822149d6d8755b5680e915a8ffec1c0.

walksanatora commented:
reverted 1 PR from upstream.
https://github.com/space-wizards/space-station-14/pull/39730

walksanatora commented:
golobby works. I can join localhost. I dont *see* any major breaking changes. so uhh LGTM >:)

Partmedia committed:
Fix benchmark (#40039)

minus1over12 committed:
Make git hooks work in git worktrees (#40038)

Make hooks work in worktrees and cleanup hooks

VerinSenpai committed:
Localize, cleanup, and LEC round control commands. (#38812)

* commit-progress

* commit

Centronias committed:
Add CVar for disabling loadout item role timers (#36775)

slarticodefast committed:
Don't show item dropping popup when wielding. (#40032)

silence

FungiFellow committed:
Trimmed Sentience Targets from Corgis Smile and Cockroaches (#39810)

* Update animals.yml

* Update pets.yml

* Removed Sentience Target from corgi

* Update random-sentience.ftl

* Obliterated Scurret from .ftl

PJBot committed:
Automatic changelog update

quasr-9 committed:
Updated inspector description to reflect functionality (#40072)

update inspector desc.

Co-authored-by: Quasr <~182430031+quasr-9@users.noreply.github.com>

Road-Train commented on Resources/Prototypes/GameRules/events.yml:
```diff
@@ -30,16 +30,41 @@
     - id: SleeperAgents
     - id: ZombieOutbreak
     - id: LoneOpsSpawn
-    - id: DerelictCyborgSpawn
```
Remove this in https://github.com/ss14Starlight/space-station-14/blob/Starlight/Resources/Prototypes/_StarLight/GameRules/events.yml

Road-Train commented on Resources/Prototypes/GameRules/events.yml:
```diff
@@ -30,16 +30,41 @@
     - id: LoneOpsSpawn
-    - id: DerelictCyborgSpawn
     - id: WizardSpawn
+    - !type:NestedSelector
```
Add these 2 lines instead.

walksanatora replied:
thats... not how it works. the table here is a weighted random. the table on the OTHER rule is a list of entity protos. you cant sub-select a weighted random table in a list of entity protos.

Road-Train commented on Resources/Prototypes/GameRules/events.yml:
```diff
@@ -30,16 +30,41 @@
     - id: SleeperAgents
     - id: ZombieOutbreak
     - id: LoneOpsSpawn
-    - id: DerelictCyborgSpawn
```
Remove this in https://github.com/ss14Starlight/space-station-14/blob/Starlight/Resources/Prototypes/_StarLight/GameRules/events.yml

walksanatora replied:
fixed

Road-Train reviewed (changes requested):
Steps to fix the yaml linter.

PubliclyExecutedPig commented:
> reverted 1 PR from upstream. [space-wizards#39730](https://github.com/space-wizards/space-station-14/pull/39730)

Why? It was a bug fix

Road-Train commented:
> > reverted 1 PR from upstream. [space-wizards#39730](https://github.com/space-wizards/space-station-14/pull/39730)
> 
> Why? It was a bug fix

Several reasons.
-1 Borgs/AI cannot unlock the upload console. So the entire point of the PR was invalid. Unsure about wizden but that's the case here, so yeah.
-2 Xenoborgs here will be getting the xenoborg lawset board to override AI's laws (the engiborg has an access breaker from wizden). So reverting this PR was necessary to retain this functionality.

PubliclyExecutedPig commented:
> > > reverted 1 PR from upstream. [space-wizards#39730](https://github.com/space-wizards/space-station-14/pull/39730)
> > 
> > 
> > Why? It was a bug fix
> 
> Several reasons. -1 Borgs/AI cannot unlock the upload console. So the entire point of the PR was invalid. Unsure about wizden but that's the case here, so yeah. -2 Xenoborgs here will be getting the xenoborg lawset board to override AI's laws (the engiborg has an access breaker from wizden). So reverting this PR was necessary to retain this functionality.

Gotcha

TiniestShark committed:
Helm + Mask Displacements for Reptilians (and some unique helmets) (#39351)

* Liz displacements + unique helms

* small fix

* Couple of small fixes

TiniestShark committed:
Reptilian tail sprites for hard/softsuits (#35842)

* Reptilian helmets and tail sprites for hard/softsuits (and a few others)!

* Removed the new fins from helmets.

* don't know how the caustic resistances got doubled? oops

* Update pirate helm sprites for lizards

* fixin errors

* ugh

* softsuit updates

* okay maybe this works now

* Corgi fixes. Boy it sure would be nice to not have to do this again.

* please work

* Helmets removed, will be done in another PR.

* missed a few.

* evil punctuation killed

* scream

* Re-parented clothingvisuals.

* Suits modified with tails

* oops chaplain tail was misnamed

* a

* one more time

* Update Resources/Prototypes/Entities/Clothing/OuterClothing/suits.yml

* Update Resources/Prototypes/Entities/Clothing/OuterClothing/suits.yml

* Update Resources/Prototypes/Entities/Clothing/OuterClothing/suits.yml

* Update Resources/Prototypes/Entities/Clothing/OuterClothing/suits.yml

* Update Resources/Prototypes/Entities/Clothing/OuterClothing/bio.yml

* Update Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml

* Update Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml

* oops chaplain lost their suit

* Small fix

* roll back unrelated fix

---------

Co-authored-by: Errant <35878406+Errant-4@users.noreply.github.com>

PJBot committed:
Automatic changelog update

TrixxedHeart committed:
Some more vox customization (#39083)

* Added docked and spiked tails, added talisman for vox

* Tail Talisman ftl

* adjusts talisman to be its own tail to avoid having to deal with other tails at all

* vox tail stuff

* Added amputated tail (courtesy of flareguy) and split tail

* adds TODO regarding marking conditionals

* rsi run check pls stop screaming at me

* Shelving the talisman for later, keeping the sprites around

* Add RSI credits

* Remove unnecessary files

* renamed amputated to vestigial

PJBot committed:
Automatic changelog update

TiniestShark committed:
Add inhand sprites to Cartons and Cups, give new inhands to Cans. (#39814)

* Added inhand sprites to cans, cartons and cups.

* small tweak

SlamBamActionman committed:
Fix small issues with Text Highlighting (#38144)

TiniestShark committed:
Burger Inhands (#39894)

burger is in your hands

MissKay1994 committed:
Adjust bureaucratic error to prevent only passenger being available (#40001)

Fire all the interns

VerinSenpai committed:
Cleanup AddPolymorphActionCommand and LEC conversion. (#38853)

commit

PJBot committed:
Automatic changelog update

SlamBamActionman committed:
Add senior courier PDA for cargo techs (#37661)

PJBot committed:
Automatic changelog update

opl- committed:
Use a fixed amount of decimal points in gas analyzer window (#40081)

This prevents the volume and temperature labels from changing width when the value lands on integer values, making the text easier to read.

Co-authored-by: opl <4833621+opl@users.noreply.github.com>

breeplayx3 committed:
Berry Delight recipe edit (#40085)

berry delight change

PJBot committed:
Automatic changelog update

sowelipililimute committed:
Don't enqueue construction events without validation (#39869)

VoidMeticulous committed:
Remove unused combat-equipped-helmet (#40095)

Remove unused combat-equipped-helmet and combat-equipped-helmet-dog

ArtisticRoomba committed:
Atmos Delta-Pressure Window Shattering (#39238)

This PR adds delta-pressure damage. In short, airtight structures can now take damage proportional to the difference in pressures between the sides of the structure.

PJBot committed:
Automatic changelog update

ArtisticRoomba committed:
Give inflatable walls the DeltaPressure component (#40098)

give inflatable walls the deltapressure component

PJBot committed:
Automatic changelog update

walksanatora committed:
fix the derelicts in tables.

Quantum-cross committed:
Add heat distortion shader for hot gases (#39107)

PJBot committed:
Automatic changelog update

jkwookee committed:
Fix QM Golden Knuckledusters not being a objective (#40096)

test

PJBot committed:
Automatic changelog update

DDeegan committed:
Telepad Label Fix (#39975)

Fixed telapad order labels showing wrong account

PJBot committed:
Automatic changelog update

Samuka-C committed:
Fool players with decoy presets (#40053)

* added secret gamepresets

* cut down on alias

* remove all secret presets

* change the command to allow for a secret argument

* update test

* moved the secret argument after the number of rounds argument

* added completions

* localization and use of CompletionHelper.Booleans

* command now has a option for a decoy preset

* fixed decoy message in the end

* ops

* clean up

* hint 2

* improve localization

---------

Co-authored-by: Errant <35878406+Errant-4@users.noreply.github.com>

PJBot committed:
Automatic changelog update

Winkarst-cpu committed:
Fix: Ability to open AHelp in the lobby by pressing the hotkey (#39525)

* Fix

* Update

VoidMeticulous committed:
Add 2.25 second delay to scurret petting (#40097)

Add 2.25 second delay to scurret petting rate

PJBot committed:
Automatic changelog update

VasilisThePikachu committed:
Stop packaging `Resources/ServerInfo` and `Resources/Changelog` on the server (#39897)

* Stop packaging `Resources/ServerInfo`

This is only used by the client, it is unneccery to pack into the server. Plus it keeps getting people to think that just editing the server resources will modify the guidebook even though that needs a custom dev enviroment.

* Add credits too

PJB3005 committed:
Package win-arm64 and osx-arm64 servers (#40113)

EmoGarbage404 committed:
Improve Gas Yaml Serialization (#40070)

* Make yaml gas serialization cleaner

* fix exception

* fix validation code

* rudimentary test & permissive loading

* change it a bit

* Test fixes and adjustments

iaada committed:
Organize JobIconPrototype yml (#39774)

grouped

Co-authored-by: iaada <iaada@users.noreply.github.com>

Quantum-cross committed:
Dynamic anomaly scanner texture (#37585)

PJBot committed:
Automatic changelog update

ScarKy0 committed:
Reuse lathe queue instead of redrawing (#39886)

* init

* init

* PUSH!!!

* //

* Me when the when the me when the

* review

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

PJBot committed:
Automatic changelog update

PJB3005 committed:
Skip MapImages folder in packaging (#38928)

Princess-Cheeseballs committed:
Can't crawl over counters (#40099)

Counters and Tables separation

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

PJBot committed:
Automatic changelog update

CawsForConcern committed:
Fixes Theobromine missing from Iced Coffee (#40063)

adding ice to coffee no longer removes its theobromine

PJBot committed:
Automatic changelog update

iaada committed:
Fix xenoborg action icons (#40118)

commit

Co-authored-by: iaada <iaada@users.noreply.github.com>

SpaceLizard24 committed:
Fixed a error in the "Adventures of Ian and Renault" books (#39932)

Fixed a publication error in the "Adventures of Ian and Renault" book series.

The editor of these books have been taken out back and shot.

ArtisticRoomba committed:
Fix DeltaPressure damage not capping beyond a certain pressure (#40125)

PJBot committed:
Automatic changelog update

ArtisticRoomba committed:
Give shutters the DeltaPressure component (#40126)

PJBot committed:
Automatic changelog update

Princess-Cheeseballs committed:
Alerts Cleanup and API (#39544)

* alert cleanup and API

* I expect update loops to be at the top.

* Address review

* Address review x 2

* Merg my PR

* Fix

* Update Content.Shared/Alert/AlertsSystem.cs

webedit

Co-authored-by: Perry Fraser <perryprog@users.noreply.github.com>

* FIX THAT TEST FAIL!!!!

* Me when I forget to actually give you alerts

* Hammedborgar

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>
Co-authored-by: Perry Fraser <perryprog@users.noreply.github.com>

Commit:
fix a typo in the comments for game.ipintel_exempt_playtime (#40129)

ScarKy0 committed:
Vulpkanin Species (#37539)

* [April Fools] Elf species  (#35353)

elf

* [April Fools] Juice that makes me go insane (#35370)

* kill me

* MAKE IT STOP

* is it finally over?

* web edit

* webedit

Co-authored-by: ArtisticRoomba <145879011+ArtisticRoomba@users.noreply.github.com>

* webedit

Co-authored-by: ArtisticRoomba <145879011+ArtisticRoomba@users.noreply.github.com>

* webedit

Co-authored-by: ArtisticRoomba <145879011+ArtisticRoomba@users.noreply.github.com>

* webedit

* WEBEDIT PLEASE

Co-authored-by: ArtisticRoomba <145879011+ArtisticRoomba@users.noreply.github.com>

---------

Co-authored-by: ArtisticRoomba <145879011+ArtisticRoomba@users.noreply.github.com>

* vulpkanin species

the ultimate april fools joke

Co-Authored-By: Kr8art <188977876+kr8art@users.noreply.github.com>

* cant be bothered

* make build and test debug run on the april fools branch (#35396)

welp

* Revert "[April Fools] Juice that makes me go insane (#35370)"

This reverts commit fcbdcf8452cb1247733242aa44ea5b9f38dfa3d2.

* Revert "[April Fools] Elf species  (#35353)"

This reverts commit 704293a051033026bfca0c9e161d97796bc0cb81.

* nuh uh

* localized names

* fix survival box

* abilities (Mostly don't work yet, pending PRs)

* wagging loc

* slight heat tweaks

* stuff

* Update Vulpkanin.xml

* vulp screams + quieter howl

* MessyDrinker by Orks

* MessyDrinker guidebook

* animal bones

* fix wagging

* missed

* no loc

* Lower damage, remove butchering

* Revert "animal bones"

This reverts commit 3457cde13e57336678f62a20bf57fb315844c240.

* leap

* collision

* leap desc update

* leap guidebook

* fix leap

* microbalance

* comment

* microbalancing pt 2

* microbalancing pt 3

* test fail fix (i hope)

* fix organs

* no leaping on the ground

* remove SolutionScanner + ling work

* fix clone

* add stuff back

* microbalancing pt yes

* male vulp names

mainly slavic names with some similiar sounding ones mixed in

* Revert "male vulp names"

This reverts commit 1666f81821bfea241262627160a594dd1b0ce050.

* male vulps names pr 2 (slavic, dog names, few puns)

Slavic names, with mixed in typical dog names and some puns

* female names (mainly slavic, other stuff mixed in)

* oops

* Update vulpkanin.yml

* Vulps WIP Work (#1)

* Modified aesprite tools

* WIP

* wipperoni

* We are done here, wawa

* Revert "Merge branch 'aseprite-tools' into hannah/vulps-wip"

This reverts commit d258645df60a94d0217fecd85a38545f2b951cde, reversing
changes made to 0ae39e862152ef6548533eba0547709594c55e90.

* Resolve merge issue

* Fix Urist McVulp forgetting his fursuit

---------

Co-authored-by: Southbridge-fur <southbridgefur@gmail.com>

* re-add ears and husk overlay

* revert adding ears to vox

* Crest Markings

* Fade Markings

* Sharp Markings

* microbalance + comments

* censor character nudity option compliance

* Ear Markings

* 1984 wagging animations

* Tail Markings

* tail wag icon

* annoyance

* guidebook

* slight temperature tweak

* seperate out snouts, port over 2 markings for testing

* fix

* seperate snout file

* Snout/Head Markings

* cleanup + locale for ears

* Head and Snout loc

* gray eyes

* crest and husk cleanup

* Tail and Chest loc

* Legs and Arms loc

* most important commit

* mime vulps like rations too

* missing loc

* microbalance

* not needed

* goatee

* white eyes

* harmony feedback tail fixes

* personal changes and fixes after playing

* fix fluffy tail clipping

* Sprite fixes, displacement fixes (#2)

* Tune head displacement

* Vulps Sprite Tuning

* helmets

* detail

* Urist

* wag

* comban't

* fix attributions

* remove vulp biosuits

* fix vulp beards

* fixed up ears

* comment note

* further ear tweaks

* engi helmets

---------

Co-authored-by: Ed <96445749+TheShuEd@users.noreply.github.com>
Co-authored-by: ArtisticRoomba <145879011+ArtisticRoomba@users.noreply.github.com>
Co-authored-by: Milon <milonpl.git@proton.me>
Co-authored-by: Kr8art <188977876+kr8art@users.noreply.github.com>
Co-authored-by: Hannah Giovanna Dawson <karakkaraz@gmail.com>
Co-authored-by: Southbridge-fur <southbridgefur@gmail.com>

PJBot committed:
Automatic changelog update

ElectroJr committed:
Rejig LogStringHandler (#30706)

* Rejig LogStringHandler

* Fix session logs

* Fix properly

* comments

* IAsType support

* Fix mind logs

* Fix mind logging AGAIN

---------

Co-authored-by: PJB3005 <pieterjan.briers+git@gmail.com>

IProduceWidgets committed:
Expedite gender reassignment (#36894)

PJBot committed:
Automatic changelog update

MidZik committed:
Adjusted minimumPlayers for Wizard midround events. (#38424)

* Adjusted minimumPlayers for Ninja and Wizard midround events

Wizard from 10->20
Ninja from 30->20
20 players matches Dragon and Loneop.

* Wizard midround minimum players from 20 to 30

* Update Resources/Prototypes/GameRules/events.yml

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

---------

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

PJBot committed:
Automatic changelog update

Tayrtahn committed:
Add water cooler interaction test (#39612)

* Add altInteract option to interaction test helper methods

* Add water cooler interaction test

* Oops, that's not a NetEntity

* Is.Not.Empty

* SPlayer

* Assert.Multiple

* Document parameters

VasilisThePikachu committed:
Add a space in osx-arm64 to fix arm64 osx builds (#40137)

PJB3005 committed:
Fix admin logs going to admin chat (#40141)

Oops

IProduceWidgets committed:
Drink outta da toiler (#40133)

* totally unrelated to any other recent additions

* dispare

* enum

PJBot committed:
Automatic changelog update

iaada committed:
Merge remote-tracking branch 'upstream/master' into up/drinks-yml

# Conflicts:
#	Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_fun.yml

iaada committed:
update for glue/lube tube inhand

ScarKy0 committed:
Disable vulpkanin human hair (#40144)

init

Commit:
Fix bad loop in LogStringHandler.AddFormat (#40147)

ArtisticRoomba committed:
Fix exo burn chamber (#40152)

PJBot committed:
Automatic changelog update

ArtisticRoomba committed:
Fix resin windows inheriting wrong dP values (#40151)

Fix resin windows inheriting from regular windows

PJBot committed:
Automatic changelog update

Princess-Cheeseballs committed:
[HOTFIX] Fix Burgers (#39773)

* Borgar

* Review

* Predicted queuedel

* Predict

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

Princess-Cheeseballs committed:
[HOTFIX] Fix Burgers (#39773)

* Borgar

* Review

* Predicted queuedel

* Predict

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

Princess-Cheeseballs committed:
[Hotfix] Wizard Rod doesn't gib the wizard. (#40041)

* Title

* Tired

* That shit did nothing goddamn

* Fix for real

* Use og code

* Hmmm borgaer

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

Princess-Cheeseballs committed:
[Hotfix] Wizard Rod doesn't gib the wizard. (#40041)

* Title

* Tired

* That shit did nothing goddamn

* Fix for real

* Use og code

* Hmmm borgaer

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

Princess-Cheeseballs committed:
Stable Into Master (#40155)

ScarKy0 committed:
Revert antique laser and appraisal tool sizes (#40158)

* init

* appraisal antique pistol sounds kinda cool tho ngl

* actually fuck items

PJBot committed:
Automatic changelog update

ToastEnjoyer committed:
Fixed disconnected grid on box station (#40161)

Fixed tesla area on box

PJBot committed:
Automatic changelog update

Dutch-VanDerLinde committed:
Add some alternate jumpsuit designs which can be toggled (#31213)

* inital

* testfa

* New Sec sprite

* Update atmos

* Update meta.json

* Update meta.json

* 0

* Update meta.json

PJBot committed:
Automatic changelog update

walksanatora committed:
merge starlight-dev into upstream-merge

walksanatora committed:
Still the worst change IMO.

walksanatora committed:
*plink* and that should be test fixed

github-actions[bot] committed:
Update Credits (#40187)

Co-authored-by: PJBot <pieterjan.briers+bot@gmail.com>

walksanatora committed:
mabey now all the duplicates are gone

walksanatora committed:
another duplicate bites the dust

walksanatora committed:
another duplicate bites the dust.

ArtisticRoomba committed:
Atmospherics Delta-Pressure YAML refactor (#40174)

ArtisticRoomba committed:
fix: Atmos dP Window Inheritance (#40192)

CroilBird committed:
Restore transfer amounts on regular syringes to 5, 10, 15 (#40197)

add transferamounts to regular syringes

PJBot committed:
Automatic changelog update

FairlySadPanda committed:
Allow Vulps With Human Hair To Be Shaved Without Clyde Joining The Circus (#40171)

* Revert "Disable vulpkanin human hair (#40144)"

This reverts commit d02aa1a4e2e106b9bfd8e9516464c9dbd86df7ca.

* You can once again shave your pet Vulp

* I can see the ass, I'm safe

* Rectified docstring as I am a good person

* I am doing this instead of playing Silksong please help

* Fix forgetting to re-add shader overriding

PJBot committed:
Automatic changelog update

chromiumboy committed:
Bug fix for APCPowerReceiverBattery (#40188)

Initial commit

ArtisticRoomba committed:
Atmos dP Guidebook Entry (#40194)

* Add Atmos dP guidebook

* Update Resources/ServerInfo/Guidebook/Engineering/DeltaPressure.xml

---------

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

PJBot committed:
Automatic changelog update

HeyBanditoz committed:
Being grappled with a grapple gun allows you to cross chasms (#39983)

* Being grappled with a grapple gun allows you to cross chasms

Closes #31698

* Update Content.Shared/Weapons/Misc/SharedGrapplingGunSystem.cs

* AAAAAAAAAAAAAAAAA

---------

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>
Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

PJBot committed:
Automatic changelog update

walksanatora committed:
fuck it whole upstream file

OrbitSystem07 committed:
Lets diona sap trigger artifact blood nodes

keronshb committed:
DoAfter support for Actions (#38253)

* Adds Action DoAfter Events

* Adds DoAfterArgs fields to DoAfterComp

* Adds a base doafter action

* Adds Attempt action doafter logic

* Adds doafter logic to actions

* Changes Action Attempt Doafter and action doafter to take in Performer and the original use delay. Use delay now triggers when a repeated action  is cancelled.

* Readds the TryPerformAction method and readds request perform action into the action doafter events

* Adds a force skip to DoAfter Cancel so we can skip the complete check

* Adds a Delay Reduction field to the comp and to the comp state

* Fixes doafter mispredict, changes doafter comp check to a guard clause, sets delay reduction if it exists.

* Cancels ActionDoAfter if charges is 0

* Serializes Attempt Frequency

* Comment for rework

* Changes todo into a comment

* Moves doafterargs to doafterargscomp

* Adds DoAfterArgs comp to BaseDoAfterAction

* Removes unused trycomp with actionDoAfter

* Replaces DoAfterRepateUseDelay const with timespan.zero

* Removes unused usings

* Makes SharedActionsSystem partial, adds DoAfter partial class to ActionSystem, moves ActionDoAfter logic to the SharedActionsSystem.DoAfter class

* Cleanup and prediction

* Renames OnActionDoAfterAttempt to OnActionDoAfter, moves both to Shared Action DoAfter

* Removes ActionAttemptDoAfterEvent and moves its summaries to ActionDoAfterEvent. Converts OnActionDoAfterAttempt into TryStartActionDoAfter

* Removes Extra check for charges and actiondoafters

* Sloptimization

* Cleanup

* Cleanup

* Adds param descs

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

SlamBamActionman committed:
Lets diona sap trigger artifact blood nodes (#40211)

PJBot committed:
Automatic changelog update

Happyrobot33 committed:
Merge branch 'starlight-dev' into upstream-merge

Centronias committed:
Derelict Mediborgs can Scan Solutions and see Mob Health (#40206)

Add mediborg unique components to derelict mediborg

PJBot committed:
Automatic changelog update

Kittygyat committed:
Reworded the Galoshes description to be more clear about what they actually do. (#40200)

Reworded the Galoshes description to be more clear about what they do

PJBot committed:
Automatic changelog update

PJB3005 committed:
Fix usages of TryIndex() (#39124)

* Fix usages of TryIndex()

Most usages of TryIndex() were using it incorrectly. Checking whether prototype IDs specified in prototypes actually existed before using them. This is not appropriate as it's just hiding bugs that should be getting caught by the YAML linter and other tools. (#39115)

This then resulted in TryIndex() getting modified to log errors (https://github.com/space-wizards/RobustToolbox/commit/94f98073b07bd3fa3133ae6799b34d90f46f467e), which is incorrect as it causes false-positive errors in proper uses of the API: external data validation. (#39098)

This commit goes through and checks every call site of TryIndex() to see whether they were correct. Most call sites were replaced with the new Resolve(), which is suitable for these "defensive programming" use cases.

Fixes #39115

Breaking change: while doing this I noticed IdCardComponent and related systems were erroneously using ProtoId<AccessLevelPrototype> for job prototypes. This has been corrected.

* fix tests

---------

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

Winkarst-cpu committed:
Cleanup: Remove unnecessary ``IEntityManager`` reference from the ``EmotesUIController`` (#40243)

Cleanup

Ser1-1y committed:
Make location in crew monitoring console localizable (#40247)

Centronias committed:
Make Foldable Clothing Hidden Layers "reset" Hidden Layers when un/Folding (#40251)

foldable clothing hidden layers fix

Fildrance committed:
feat: SimpleRadial menu support for sprite-view and more extensibility (#39223)

Deerstop committed:
Decal spawners spawn on a higher layer (#39956)

changed decal spawn layer

Steffo99 committed:
Fix RGB staff not working (#40258)

Add missing `TargetAction` to `ActionRgbLight`

This fixes the RGB staff not working.

PJBot committed:
Automatic changelog update

Ser1-1y committed:
Make "Confirm" in VerbMenuUIController localizable (#40248)

SlamBamActionman committed:
Add support for contraband text to the reagent guidebook (#37113)

* Add contraband text to reagent guidebook

* Add reagent for examining

* Update Content.Client/Guidebook/Controls/GuideReagentEmbed.xaml.cs

---------

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

Winkarst-cpu committed:
Fix emergency evac shuttle console early launch mispredict (#39751)

* Fix

* Yes

* Mess

* Update

* Like that?

iaada committed:
SpawnEntityTableOnTrigger (#39909)

* commit

* comment

* empty

* better xform

---------

Co-authored-by: iaada <iaada@users.noreply.github.com>

iaada committed:
PopupOnTrigger (#39913)

* commit

* comment

* changes

* Update Content.Shared/Trigger/Systems/PopupOnTriggerSystem.cs

---------

Co-authored-by: iaada <iaada@users.noreply.github.com>
Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

Princess-Cheeseballs committed:
[HOTFIX] Fix Loadout Validations (#40189)

p0 bugfix

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

Errant-4 committed:
Merge branch 'master' into fix-conflict-40263

Errant-4 committed:
Merge Stable into Master (#40263)

insoPL committed:
Do after checks for being inside container (#39880)

fix

PJBot committed:
Automatic changelog update

PJB3005 committed:
Ignore non-content commands in AllCommandsHavePermissions (#39336)

Causing a test failure every time a Toolshed command gets added to engine is ridiculous.

c4llv07e committed:
Allow to run `mappingclientsidesetup` and `showsubfloor` with +MAPPING permissions (#34455)

Allow +MAPPING admins to call mappingclientsidesetup and showsubfloor

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

SweetAplle committed:
"idk" no longer shrugs, instead sanitizing to "I don't know" (#39024)

* idk no longer shrugs

* accidentally had this in here!

* this was accidentally in here too!

* fucked up the removal

* fucked it up again this should HOPEFULLY be the last one

turns out it's hard to revert the edit of a file that's just numbers and spaces who woulda thunk?

* Restore SpaceStation14.sln to master

* Restore development.toml

---------

Co-authored-by: Tayrtahn <tayrtahn@gmail.com>

PJBot committed:
Automatic changelog update

Tayrtahn committed:
Improve Do Not Map test to whitelist specific prototypes per map and whitelist entire directories (#36117)

* Enable whitelisting specific DNM prototypes per map

* Enable whitelisting directories

* Rename fields

* Use a HashSet instead of an array

* Add check for unused whitelist entries

* Remove whitelisting for meta (warden's rubber stamp was removed)

* Add glob support courtesy of @IProduceWidgets

* Update xmldoc

imatsoup committed:
Make Butterflies zombie immune (#40265)

butterfly zombie no longer

PJBot committed:
Automatic changelog update

Princess-Cheeseballs committed:
Cardboard Box Weightless Fix (#40260)

Box is no longer space faring

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

PJBot committed:
Automatic changelog update

IProduceWidgets committed:
Add admin shuttles (#32139)

* dis da shuttles tho

* power checking

* rerun tests

* purge invalids

* attributions

* Update shuttle save files for new serialization.

* get regexed nerd

* fix shuttle yml guh

* Kill actions

PJBot committed:
Automatic changelog update

SlamBamActionman committed:
Exo - Lighting update & more (#40199)

PJBot committed:
Automatic changelog update

ToastEnjoyer committed:
Laser rifle is contraband again (#40253)

* Fixed contraband tag

* practice isnt contraband

* Added base

* Removal

PJBot committed:
Automatic changelog update

GeneralGaws committed:
Fix darts inhand sprites (#40207)

fix

Co-authored-by: GeneralGaws <limonmessi@mail.ru>

Absotively committed:
Fix APC breaker toggle button prediction by setting ToggleMode True (#40273)

* Use ToggleMode for toggle button

* Actually this bit doesn't require changing apparently

slarticodefast committed:
fix chasmsystem resolve error (#40281)

IProduceWidgets committed:
No take; Only throw. (#40143)

* commit atrocities

* hail satan

* Channel Pavlovian horror

* pet kitty cat

* Taint universe forever

* Assault sensibilities

* sully existence

* tarnish morality

* The degunnening

* encratening

* Update toys.yml

* add few pixels to fix vulp inhands

---------

Co-authored-by: ScarKy0 <scarky0@onet.eu>

PJBot committed:
Automatic changelog update

cmorley191 committed:
Clip the WindowTitle of FancyWindows, so close buttons don't get hidden (#40272)

ClipText of FancyWindow title, so close buttons don't get hidden

SlamBamActionman committed:
Food Item Size Adjustment (#39203)

PJBot committed:
Automatic changelog update

SlamBamActionman committed:
They're milk jugs, not milk cartons!

SlamBamActionman committed:
Fix Linter errors (#40283)

Fix yoml

ScarKy0 committed:
Remove unused BulletTennis (#40285)

init

SlamBamActionman committed:
Merge remote-tracking branch 'upstream-origin/master' into fork/iaada/more-trash

SlamBamActionman committed:
Reorganize and refactor drinks yml (#39221)

PJBot committed:
Automatic changelog update

perryprog committed:
Make vending machine restocks predicted (and its sound not spammable) (#38609)

* feat: make vending machine restocks predicted

* refactor: VendingMachineRestockComponent cleanup

* refactor: minor simplification

* revert: refactor: minor simplification; load bearing IsFirstTimePredicted

lol second guessed myself

* chore: unneeded VendingMachineSystem dep

* Update Content.Shared/VendingMachines/VendingMachineComponent.cs

---------

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

luegamer committed:
Added SmartFridge circuitboards (#39879)

PJBot committed:
Automatic changelog update

UpAndLeaves committed:
storage and inventory toolshed commands (#39046)

* First commit

* CommandImplementation on singletons is die, a la moony

* Fix duplicated thingy because yes

* Prototypes, bugfixes, refactoring oh my

* Remember to actually stage your ftl changes next time, leaf

PJBot committed:
Automatic changelog update

slarticodefast committed:
move all the radio components and system to Shared (#40293)

* move all the radio components and system to Shared.

* duh split impl

* address reviews

* cleanup

---------

Co-authored-by: walksanatora <walkerffo22@gmail.com>

Archeonan committed:
Clake frag round fix (#40294)

make the frag work

insoPL committed:
Fix for can't stop pulling when cuffed (#40233)

* fix

* fix

* spaces added

TheShuEd committed:
Stop microwaving! (#40132)

* Create KillMicrowaveTest.cs

* Update KillMicrowaveTest.cs

* Update Content.IntegrationTests/Tests/Microwave/KillMicrowaveTest.cs

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

* Update Content.IntegrationTests/Tests/Microwave/KillMicrowaveTest.cs

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

* Update Content.IntegrationTests/Tests/Microwave/KillMicrowaveTest.cs

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

* documentation

* Apply suggestions from code review

---------

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

sowelipililimute committed:
De-enumify humanoid species skin colours (#39175)

* De-enumify humanoid species skin colours

* Change index to resolve

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

walksanatora committed:
Merge branch 'starlight-dev' into upstream-merge

walksanatora committed:
fix for event-based weightless

VasilisThePikachu committed:
Revert "Can't crawl over counters (#40099)"

This reverts commit 01a7fc66f011967579330b1c155c326b61fa5806.

@Princess-Cheeseballs

This is being reverted per maintainer vote for the release cycle, check out the meeting video here https://youtu.be/N5-UYCLha2I?t=872

slarticodefast committed:
Revert "Add heat distortion shader for hot gases" (#40352)

ScarKy0 committed:
Vulpkanin Color Clamping (#40348)

Princess-Cheeseballs committed:
[STAGING/HOTFIX] Butcher entities in containers. (#40299)

* Kitchen spike and sharp system

* Use transform and parent

* A

* Works

* A

---------

Co-authored-by: Princess Cheeseballs <66055347+Pronana@users.noreply.github.com>

Errant-4 committed:
[Hotfix] Remove pull-escape trick (#40368)

* make HandleStopPull byref

* we get signal

walksanatora committed:
Merge branch 'starlight-dev' into upstream-merge

walksanatora committed:
merge wizden/stable into upstream-merge (your taking too long)

walksanatora commented:
we took too long
now we are gettign 4 weeks of content at once
random youtube music when I made this comment go:
https://music.youtube.com/watch?v=jPvfmGKBWIE

walksanatora committed:
Merge branch 'starlight-dev' into upstream-merge

InanimateIJK commented:
havent looked thru this yet but this doesn't have the upstream vulpkanin's commits rights?

walksanatora committed:
Fix compile errors

walksanatora commented:
> havent looked thru this yet but this doesn't have the upstream vulpkanin's commits rights?

it does. I am working to sort em out right now (and appeasing yaml linter at the same time)

walksanatora committed:
appease yaml linter non-vulp

walksanatora committed:
vulp 1984

walksanatora committed:
re-adds yip. also makes their emote sounds less dog like. yes they are "generic canine" but the emote sounds are too anamalistic if you catch what I mean.

walksanatora commented:
now includes https://discord.com/channels/310555209753690112/1079439851750953082/1416898665494216714 and latter messages

walksanatora committed:
fix speech path.

walksanatora committed:
the death of mime baugette

walksanatora committed:
Not S&P approved

walksanatora committed:
fix containers

StarlightHost committed:
Merge branch 'starlight-dev' into upstream-merge
